rasa/cdu/command_processor.py

```diff
@@ -242,22 +265,30 @@ def filled_slots_for_active_flow(
     All slots that have been filled for the current flow.
     """
     flow_stack = FlowStack.from_tracker(tracker)
-
-    top_flow = flow_stack.top_flow(all_flows)
     top_flow_step = flow_stack.top_flow_step(all_flows)
 
     current_question = (
         top_flow_step.question if isinstance(top_flow_step, QuestionFlowStep) else None
     )
 
-    if top_flow_step is not None and top_flow is not None:
-        return {
-            q.question
-            for q in top_flow.previously_asked_questions(top_flow_step.id)
-            if q.question != current_question
-        }
-    else:
-        return set()
+    asked_questions = set()
+
+    for frame in reversed(flow_stack.frames):
+        if not (flow := all_flows.flow_by_id(frame.flow_id)):
+            break
+
+        for q in flow.previously_asked_questions(frame.step_id):
+            if q.question != current_question:
+                asked_questions.add(q.question)
+
+        if frame.frame_type in STACK_FRAME_TYPES_WITH_USER_FLOWS:
+            # as soon as we hit the first stack frame that is a "normal"
+            # user defined flow we stop looking for previously asked questions
+            # because we only want to ask questions that are part of the
+            # current flow.
+            continue
+
+    return asked_questions
```

Comment:
> We break when a flow that is on the stack does not exist ~ How did this flow get on the stack in the first place?
yeah we have that in waaaaay to many places. I am very inclined to change the interface of 
```
def flow_by_id(self, id: Optional[Text]) -> Optional[Flow]:
```
to this
```
def flow_by_id(self, id: Text) -> Flow:
```
and have it throw an exception if it can't look up the id. because we really ever should have an id there that doesn't exist and if there is something is off and we can raise a full blown error.

BUT writing that: we have "mock stack frames" for intentless / search...

Reply:
I just double checked, the `continue` should be a `break` - good catch :+1:

Reply:
i'll keep the interface of flow_by_id like it is for now and tackle that separately.